### PR TITLE
React docs part 9

### PR DESCRIPTION
**Pull request by BenAtEventbrite**

### CODE REVIEW

File: react/README.md [hunk 1]
```diff
@@ -1453,7 +1453,94 @@ export default class TextInput extends React.Component {
...
+
+// good
+export default class TeamPicker extends React.Component {
+    _handleTeamClick(teamId) {
+        location.href = TEAMS[teamId].url;
+    }
+
+    render() {
+        let teamButtons = Object.keys(TEAMS).map((teamId) => (
+            <button onClick={this._handleTeamClick.bind(this, teamId)} key={teamId}>
```
Comment (jphilipps-eb) [inline react/README.md]: So, this is a case where the `teamId` is a number, but it's not the `index` in the map, so it's ok to use it as the `key`?

File: react/README.md [hunk 2]
```diff
@@ -1453,7 +1453,94 @@ export default class TextInput extends React.Component {
...
+            <div>
+                <h2>Pick your team</h2>
+                <div>{teamButtons}</div>
+            </div>
+        );
+    }
+}
+```
+
+While the two examples above look very similar, the differences make all the difference.
```
Comment (jphilipps-eb) [inline react/README.md]: hmm, I wonder if there is a better way to say this? ;)

File: react/README.md [hunk 3]
```diff
@@ -1453,7 +1453,94 @@ export default class TextInput extends React.Component {
...
+
+In the _good_ example, we pass the `teamId` to the [`.bind()` method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/bind) such that when the `onClick` handler is invoked, the `teamId` is the first parameter in `_handleTeamClick`. This is accomplished with: `this._handleTeamClick.bind(this, teamId)`.
+
+ This prevents the need to store the identifier in the DOM using the `data-teamId` attribute in the _bad_ example. The reason that this is "bad" is because the code now has to unnecessarily touch the DOM in order to retrieve `teamId`, when JavaScript already has the value.
+
+ In this contrived example, the performance difference between _good_ and _bad_ is likely negligible, but at scale, a lot of unnecessary DOM accesses can really slow down an app. Always adhering to this good practice should prevent the need to go hunting for performance bottlenecks in an existing large app.
```
Comment (jphilipps-eb) [inline react/README.md]: good explanation - makes sense!

File: react/README.md [hunk 4]
```diff
@@ -1453,7 +1453,94 @@ export default class TextInput extends React.Component {
...
+
+```js
+// shared data
+const TEAMS = {
+    'warriors': {
+        name: 'Golden State Warriors',
+        url: 'http://www.nba.com/warriors'
+    },
+    '49ers': {
+        name: 'San Francisco',
```
Comment (kaylieEB) [inline react/README.md]: 49ers!

File: react/README.md (hunk 1 above)
Comment (BenAtEventbrite) [inline react/README.md]: `teamId` is the string `'warriors'`, `'49ers'`, etc. An id/identifier doesn't have to be a number. It's just a way to uniquely identify an object. Typically they are numbers, strings, [GUIDs](https://en.wikipedia.org/wiki/Globally_unique_identifier), etc.

But if it's causing confusion, I can rename. Do you have any suggestions?

File: react/README.md (hunk 2 above)
Comment (BenAtEventbrite) [inline react/README.md]: I thought it was clever! 😢

File: react/README.md (hunk 1 above)
Comment (jphilipps-eb) [inline react/README.md]: got it - I think you can leave it as it. I'm just programmed to think that `id` = number, so I will stop making that assumption

**Pull request merged**